src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -158,14 +158,16 @@ private void EmitParametersIfPresent(JsonTextWriter jsonWriter, ExpressionEmitte
             jsonWriter.WriteEndObject();
         }
 
-        private ObjectSyntax EvaluateDecorators(StatementSyntax statement, ObjectSyntax input, TypeSymbol targetType)
+        private ObjectSyntax EvaluateDecorators(StatementSyntax statement, ObjectSyntax input, TypeSymbol targetType, bool skipBatchSize = false)
```

Comment:
This feels very special-case. What about adding a hashset similar to `ResourcePropertiesToOmit`, named something like `DecoratorsToEmitAsResourceProperties` and inverting the check - e.g. something like:

```csharp
private ObjectSyntax EvaluateDecorators(StatementSyntax statement, ObjectSyntax input, TypeSymbol targetType)
{
    ...
    foreach (var decoratorSyntax in statement.Decorators.Reverse())
    {
        if (symbol is FunctionSymbol decoratorSymbol && DecoratorsToEmitAsResourceProperties.Contains(decoratorSymbol.Name))
        {
            ...
```

Reply:
Done, was a good choice